cmd/storagenode-updater/main_test.go

```diff
@@ -4,68 +4,194 @@
-		},
-		{
-			"testdata/fake-ident.cert",
-			ctx.File("identity.cert"),
-			0644,
-		},
-		{
-			"testdata/fake-ident.key",
-			ctx.File("identity.key"),
-			0600,
-		},
+	oldSemVer, err := version.NewSemVer(oldVersion)
+	require.NoError(t, err)
+
+	newSemVer, err := version.NewSemVer(newVersion)
+	require.NoError(t, err)
+
+	oldInfo := version.Info{
+		Timestamp:  time.Now(),
+		CommitHash: "",
+		Version:    oldSemVer,
+		Release:    false,
 	}
 
-	for _, file := range testFiles {
-		content, err := ioutil.ReadFile(file.src)
-		require.NoError(t, err)
+	// build real storagenode and updater with old version
+	oldStoragenodeBin := ctx.CompileWithVersion("storj.io/storj/cmd/storagenode/", oldInfo)
+	storagenodePath := ctx.File("fake", "storagenode.exe")
+	copyBin(ctx, t, storagenodePath, oldStoragenodeBin)
 
-		err = ioutil.WriteFile(file.dst, content, file.perms)
-		require.NoError(t, err)
+	oldUpdaterBin := ctx.CompileWithVersion("", oldInfo)
+	updaterPath := ctx.File("fake", "storagenode-updater.exe")
+	copyBin(ctx, t, updaterPath, oldUpdaterBin)
+
+	// build real storagenode and updater with new version
+	newInfo := version.Info{
+		Timestamp:  time.Now(),
+		CommitHash: "",
+		Version:    newSemVer,
+		Release:    false,
+	}
+	newStoragenodeBin := ctx.CompileWithVersion("storj.io/storj/cmd/storagenode/", newInfo)
+	newUpdaterBin := ctx.CompileWithVersion("", newInfo)
+	updateBins := map[string]string{
+		"storagenode":         newStoragenodeBin,
+		"storagenode-updater": newUpdaterBin,
 	}
 
-	var mux http.ServeMux
-	content, err := ioutil.ReadFile("testdata/fake-storagenode.zip")
+	// run versioncontrol and update zips http servers
+	versionControlPeer, cleanupVersionControl := testVersionControlWithUpdates(ctx, t, updateBins)
+	defer cleanupVersionControl()
+
+	logPath := ctx.File("storagenode-updater.log")
+
+	// write identity files to disk for use in rollout calculation
+	identConfig := testIdentityFiles(ctx, t)
+
+	// run updater (update)
+	args := []string{"run"}
+	args = append(args, "--config-dir", ctx.Dir())
+	args = append(args, "--server-address", "http://"+versionControlPeer.Addr())
+	args = append(args, "--binary-location", storagenodePath)
+	args = append(args, "--check-interval", "0s")
+	args = append(args, "--identity.cert-path", identConfig.CertPath)
+	args = append(args, "--identity.key-path", identConfig.KeyPath)
+	args = append(args, "--log", logPath)
+
+	// NB: updater currently uses `log.SetOutput` so all output after that call
+	// only goes to the log file.
+	out, err := exec.Command(updaterPath, args...).CombinedOutput()
+	logData, logErr := ioutil.ReadFile(logPath)
+	if assert.NoError(t, logErr) {
+		logStr := string(logData)
+		t.Log(logStr)
+		if !assert.Contains(t, logStr, "storagenode restarted successfully") {
+			t.Log(logStr)
+		}
+		if !assert.Contains(t, logStr, "storagenode-updater restarted successfully") {
+			t.Log(logStr)
+		}
+	} else {
+		t.Log(string(out))
+	}
+	if !assert.NoError(t, err) {
```

Comment:
There is no need for this assertion, just move the `t.FailNow()` inside of the above `else` block.

Reply:
done